### PR TITLE
Return srcCurMalformed for taker_pays currency in book_offers

### PR DESCRIPTION
**Pull request by officialfrancismendoza**
**Issue** (#267): book_offers API returns "invalidParams" instead of "srcCurMalformed" for taker_pays currency field
**Fix**: Add srcCurMalformed in taker_pays currency field

### CODE REVIEW

File: src/rpc/RPCHelpers.cpp
Comment (godexsoft) [inline src/rpc/RPCHelpers.cpp:1398]: Is this the same error you would get from rippled? LGTM if it is.

Comment (officialfrancismendoza) [reply src/rpc/RPCHelpers.cpp:1398]: @godexsoft this was a reused error message for a new conditional to handle the new case of XRP pay currency with an XRP specified issuer (with the latter not being necessary at all). This solves one of the cases @ManiMounikaKunasani added later

**Comment by officialfrancismendoza** [comment]
@godexsoft can you review in light of the new merge commit? If everything looks good, will squash and then ask you to approve and then merge it, as I don't have privileges to merge automatically

File: src/rpc/RPCHelpers.cpp
Comment (godexsoft) [inline src/rpc/RPCHelpers.cpp]: Why change this?

File: src/rpc/RPCHelpers.cpp
Comment (godexsoft) [inline src/rpc/RPCHelpers.cpp]: Does not look like you need this temporary but if you really want to have it you would have to `std::move()` it if you left the signature of above callback as it should be (`void(ripple::SLE&&)`).

**Review by godexsoft — CHANGES_REQUESTED** [review]

File: src/rpc/RPCHelpers.cpp
Comment (godexsoft) [inline src/rpc/RPCHelpers.cpp]: You may need to setup VSCode (or the IDE you are using) to automatically add a newline to the end of any files you change.

**Review by godexsoft — APPROVED** [review]
LGTM.

**Pull request closed without merging**
